### PR TITLE
fix: crash when saving edited PDF files

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/32476.

This commit fixes a crash that persists any time a user attempts to download an edited PDF. This was happening because the logic flow for download of any edited PDF triggers a call to `chrome.fileSystem.chooseEntry`, which we do not support and which cause unmapped page access crashes.

This patch can be removed should we choose to support `chrome.fileSystem` or support it enough to fix the crash.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed a crash that occurred when user attempted to download an edited PDF

### CODE REVIEW

**Review by miniak — APPROVED** [review]

File: patches/chromium/fix_crash_when_saving_edited_pdf_files.patch [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+--- a/chrome/browser/resources/pdf/pdf_viewer.js
++++ b/chrome/browser/resources/pdf/pdf_viewer.js
+@@ -967,25 +967,12 @@ export class PDFViewerElement extends PDFViewerBaseElement {
+       dataArray = [result.dataToSave];
+     }
+ 
++    const a = document.createElement('a');
++    a.download = fileName;
+     const blob = new Blob(dataArray);
+-    const fileName = this.attachments_[index].name;
```
Comment (jkleinsc) [inline patches/chromium/fix_crash_when_saving_edited_pdf_files.patch:26]: Looking at the patched file, it does not appear that `fileName` is defined.

**Review by jkleinsc — CHANGES_REQUESTED** [review]
It looks like the patch needs an update to make sure `fileName` is defined before its referenced.

File: patches/chromium/fix_crash_when_saving_edited_pdf_files.patch (hunk 1 above)
Comment (codebytere) [reply patches/chromium/fix_crash_when_saving_edited_pdf_files.patch:26]: ah oops i'll fix this - bad copy from the non-attachment section!

Comment (codebytere) [reply patches/chromium/fix_crash_when_saving_edited_pdf_files.patch]: done!

**Review by jkleinsc — APPROVED** [review]

**Review by nornagon — COMMENTED** [review]
Can you try upstreaming this? The `chrome.fileSystem` API is apparently [deprecated](https://developer.chrome.com/docs/extensions/reference/fileSystem/), so upstream might be interested in help moving off it :)

**Comment by codebytere** [comment]
@nornagon done: https://chromium-review.googlesource.com/c/chromium/src/+/3397981

**Pull request merged**